internal/pkg/cli/pipeline_update.go

```diff
@@ -0,0 +1,245 @@
+	fmtAddPipelineResourcesComplete = "Successfully added pipeline resources to your project: %s"
+
+	fmtUpdatePipelineFailed   = "Failed to accept changes for pipeline: %s."
+	fmtUpdatePipelineStart    = "Proposing infrastructure changes for the pipeline: %s"
+	fmtUpdatePipelineComplete = "Successfully updated pipeline: %s"
+)
+
+var errNoPipelineFile = errors.New("There was no pipeline manifest found in your workspace. Please run `archer pipeline init` to create an pipeline.")
+
+// UpdatePipelineOpts holds the configuration needed to create or update a pipeline
+type UpdatePipelineOpts struct {
+	// Fields with matching flags.
+	PipelineFile string
+	// Deploy bool
+
+	pipelineDeployer pipelineDeployer
+	project          *archer.Project
+	prog             progress
+	region           string
+	envStore         archer.EnvironmentStore
+	account          string
+	ws               archer.Workspace
+
+	*GlobalOpts
+}
+
+func NewUpdatePipelineOpts() *UpdatePipelineOpts {
+	return &UpdatePipelineOpts{
+		GlobalOpts: NewGlobalOpts(),
+		prog:       termprogress.NewSpinner(),
+	}
+}
+
+// func (opts *UpdatePipelineOpts) Ask() error {
+// 	if opts.PipelineFile() == "" {
+// 		return errNoPipelineFile
+// 	}
+
+// 	return nil
+// }
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (opts *UpdatePipelineOpts) Validate() error {
+	if opts.PipelineFile == "" {
+		return errNoPipelineFile
+	}
+
+	return nil
+}
+
+func (opts *UpdatePipelineOpts) convertStages(manifestStages []manifest.PipelineStage) ([]deploy.PipelineStage, error) {
+	stages := []deploy.PipelineStage{}
+	apps, err := opts.ws.AppNames()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, stage := range manifestStages {
+		env, err := opts.envStore.GetEnvironment(opts.project.Name, stage.Name)
+		if err != nil {
+			return nil, err
+		}
+
+		pipelineStage := deploy.PipelineStage{
+			LocalApplications: apps,
+			AssociatedEnvironment: &deploy.AssociatedEnvironment{
+				Name:      stage.Name,
+				Region:    env.Region,
+				AccountID: env.AccountID,
+				Prod:      env.Prod,
+			},
+		}
+		stages = append(stages, pipelineStage)
+	}
+
+	return stages, nil
+}
+
+func (opts *UpdatePipelineOpts) getArtifactBuckets() ([]deploy.ArtifactBucket, error) {
+	regionalResources, err := opts.pipelineDeployer.GetRegionalProjectResources(opts.project)
+	if err != nil {
+		return nil, err
+	}
+
+	var buckets []deploy.ArtifactBucket
+	for _, resource := range regionalResources {
+		bucket := deploy.ArtifactBucket{
+			BucketName: resource.S3Bucket,
+			KeyArn:     resource.KMSKeyARN,
+		}
+		buckets = append(buckets, bucket)
+	}
+
+	return buckets, nil
+}
+
+func (opts *UpdatePipelineOpts) Execute() error {
+	// bootstrap pipeline resources
+	opts.prog.Start(fmt.Sprintf(fmtAddPipelineResourcesStart, color.HighlightUserInput(opts.project.Name)))
+	err := opts.pipelineDeployer.AddPipelineResourcesToProject(opts.project, opts.region)
+	if err != nil {
+		return nil
+	}
```

Comment:
Are the errors returned from `AddPipelineResourcesToProject` clear to the user? or should we wrap it with additional context? (same comment for other errors returned from public functions)

Reply:
Not sure, but we can handle that in a future PR for general UX improvements (i know a guy ;) )